### PR TITLE
Add array_combine dynamicExtension

### PR DESCRIPTION
**Pull request by VincentLanglet**
Close https://github.com/phpstan/phpstan/issues/1580

### CODE REVIEW

**Comment by VincentLanglet** [comment]
I don't understand the failure @ondrejmirtes

File: src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		return $functionReflection->getName() === 'array_combine';
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		if (count($functionCall->args) < 2) {
+			return ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType();
+		}
+
+		$args0 = $functionCall->args[0]->value;
```
Comment (ondrejmirtes) [inline src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php:41]: Please rename to $firstArg

File: src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		$args1 = $functionCall->args[1]->value;
```
Comment (ondrejmirtes) [inline src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php:42]: $secondArg

File: src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+				);
+			}
+		}
+
+		$arrayType = new ArrayType(
+			$keysParamType instanceof ArrayType ? $keysParamType->getItemType() : new MixedType(),
+			$valuesParamType instanceof ArrayType ? $valuesParamType->getItemType() : new MixedType()
+		);
+
+		if ($this->phpVersion->getVersionId() >= 80000) {
```
Comment (ondrejmirtes) [inline src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php:72]: Please add a describing method to PhpVersion class and use it here.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php (hunk 3 above)
Comment (VincentLanglet) [reply src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php]: I created `throwsTypeErrorForInternalFunctions`

File: src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php (hunk 2 above)
Comment (VincentLanglet) [reply src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php]: Done

File: src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Type/Php/ArrayCombineFunctionReturnTypeExtension.php]: Done

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**